### ISSUE TITLE
Train neural network

### ISSUE THREAD

**Issue by DanielEggelmann** [sagr4019/ResearchProject#143]
Train the LSTM neural network with the revised generated and tokenized code.

**Comment by DanielEggelmann** [comment]
Hyperparameters:
- training samples: 36000 (only samples of a size of 100 tokens and lower)
- seed: 123456789
- epochs: 29
- batch size: 64
- LSTM layers: 1
- LSTM units: 256
- embedding dimensions: 32
- learning algorithm: Adam
- loss: "categorical_crossentropy"

Results:
- Accuracy: 0.99
- Loss: 0.0115
- Average training time per epoch: 1:45 minutes

**Comment by jan-christiansen** [comment]
Please add instructions for reproduction to the readme.

**Comment by DanielEggelmann** [comment]
Retraining with fixed codegenerator.
Hyperparameters:
- training samples: 26000 (only samples of a size of 50 tokens and lower)
- validation samples: 2600
- seed: 123456789
- epochs: 50
- batch size: 64
- LSTM layers: 1
- LSTM units: 256
- embedding dimensions: 32
- learning algorithm: Adam
- loss: "categorical_crossentropy"

Results:
- Accuracy: 0.97
- Loss: 0.08
- Average training time per epoch: 37 seconds
